src/main/java/cz/cvut/kbss/termit/service/document/backup/DocumentBackupManager.java

```diff
@@ -0,0 +1,240 @@
+            return tempFile.toFile();
+        } catch (IOException | CompressorException | NullPointerException e) {
+            throw new BackupManagerException("Unable to decompress file.", e);
+        }
+    }
+
+    /**
+     * Compresses the given file with BZip2 into the same folder
+     * @param file the file to compress
+     * @implSpec The file cannot be larger than 2GB
+     * @see #decompressFile(java.io.File)
+     */
+    private void compressFile(java.io.File file) {
+        if (!file.isFile()) {
+            throw new BackupManagerException("Unable to compress non-file entry: " + file.getAbsolutePath());
+        }
+        String compressedFileName = BZip2Utils.getCompressedFileName(file.getName());
+        Path compressedFilePath = file.toPath().getParent().resolve(compressedFileName);
+        try (
+                // final bzip2 compressed file
+                OutputStream fos = Files.newOutputStream(compressedFilePath);
+                // bzip2 writes to ↑
+                CompressorOutputStream<?> cos = new CompressorStreamFactory().createCompressorOutputStream(CompressorStreamFactory.BZIP2, fos);
+                InputStream is = Files.newInputStream(file.toPath())
+        ) {
+            IOUtils.copy(is, cos);
+        } catch (IOException | CompressorException | NullPointerException e) {
+            throw new BackupManagerException("Unable to compress file.", e);
+        }
+    }
+
+    /**
+     * List available backups for the specified file.
+     * @param file backed up file
+     * @param reason reason for which backups are returned, or {@code null}
+     * @return List of available backups
+     */
+    public List<BackupFile> getBackups(File file, BackupReason reason) {
+        java.io.File directory = storageDirectory.resolve(file.getDirectoryName()).toFile();
+        if (!directory.isDirectory()) {
+            return List.of();
+        }
+        final String fileName = IdentifierResolver.sanitizeFileName(file.getLabel());
+        java.io.File[] files = directory.listFiles((dir, filename) -> filename.startsWith(fileName));
+        if (files == null) {
+            return List.of();
+        }
+        Stream<BackupFile> filesStream = Arrays.stream(files).map(this::parseBackupFile).filter(Objects::nonNull);
+        if (reason != null) {
+            filesStream = filesStream.filter(b -> reason.equals(b.backupReason()));
+        }
+        return filesStream.toList();
+    }
+
+    /**
+     * Parse the name of the given file in the file system resolving timestamp and backup reason.
+     * <p>
+     * If the timestamp is missing, the current time is used.
+     * <p>
+     * If the backup reason cannot be used, {@link BackupReason#UNKNOWN UNKNOWN} is used.
+     * @param file the file to parse
+     * @return The parsed backup file or {@code null} when parsing fails
+     * when the timestamp could not be parsed.
+     */
+    private BackupFile parseBackupFile(java.io.File file) {
```

Comment:
`parseBackupFileName`? `pareBackupFile` seems a bit misleading, given we are not actually parsing the file content, just its name (and it is states so in the Javadoc)...